SimG4CMS/Muon/src/MuonSensitiveDetector.cc

```diff
@@ -94,64 +92,57 @@ MuonSensitiveDetector::~MuonSensitiveDetector() {
   delete slaveMuon;
   delete theRotation;
   delete detector;
-
   delete theG4ProcessTypeEnumerator;
-  
-  delete myG4TrackToParticleID;
 }
 
 void MuonSensitiveDetector::update(const BeginOfEvent * i){
   clearHits();
-
   //----- Initialize variables to check if two steps belong to same hit
   thePV = nullptr;
   theDetUnitId = 0;
   theTrackID = 0;
-
 }
 
-void MuonSensitiveDetector::update(const  ::EndOfEvent *)
-{}
-
-
 void MuonSensitiveDetector::clearHits()
 {
-  LogDebug("MuonSimDebug") << "MuonSensitiveDetector::clearHits"<<std::endl;
+  LogDebug("MuonSimDebug") << "MuonSensitiveDetector::clearHits";
   slaveMuon->Initialize();
 }
 
 bool MuonSensitiveDetector::ProcessHits(G4Step * aStep, G4TouchableHistory * ROhist)
 {
-  LogDebug("MuonSimDebug") <<" MuonSensitiveDetector::ProcessHits "<<InitialStepPosition(aStep,WorldCoordinates)<<std::endl;
-
- // TimeMe t1( theHitTimer, false);
+  LogDebug("MuonSimDebug") <<" MuonSensitiveDetector::ProcessHits "
+			   <<InitialStepPosition(aStep,WorldCoordinates);
 
   if (aStep->GetTotalEnergyDeposit()>0.){
+    newDetUnitId = setDetUnitId(aStep);
+
     // do not count neutrals that are killed by User Limits MinEKine
+    //---VI: this is incorrect, neutral particle, like neutron may have local
+    //       energy deposit, which potentially may make a hit
     if( aStep->GetTrack()->GetDynamicParticle()->GetCharge() != 0 ){
   
       if (newHit(aStep)) {
 	saveHit();
 	createHit(aStep);
       } else {
 	updateHit(aStep);
-      }    
-      return true;
+      } 
     } else {
-      storeVolumeAndTrack(aStep);
-      return false;
+      thePV = aStep->GetPreStepPoint()->GetPhysicalVolume();
+      theTrackID = aStep->GetTrack()->GetTrackID();
+      theDetUnitId = newDetUnitId;
     }
   }
-  return false;
+  return true;
```

Comment:
@civanch for my understanding, what is implied by this change of logic? What was the triggered action, if any, by a "false" value returned?

Reply:
There is no difference to me - the returned bool does not used anywhere. I put "true" in order not make impression to newcomer, that a choice should be done between "true" and "false".